test/TestProjects/MgmtVirtualResource/Generated/VirtualMachineMgmtVirtualResource.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+
+namespace MgmtVirtualResource
+{
+    /// <summary>
+    /// A Class representing a VirtualMachine along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="VirtualMachineMgmtVirtualResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetVirtualMachineMgmtVirtualResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ResourceGroupResource" /> using the GetVirtualMachine method.
+    /// </summary>
+    public partial class VirtualMachineMgmtVirtualResource : ArmResource
+    {
+        /// <summary> Generate the resource identifier of a <see cref="VirtualMachineMgmtVirtualResource"/> instance. </summary>
+        public static ResourceIdentifier CreateResourceIdentifier(string subscriptionId, string resourceGroupName, string vmName)
```

Comment:
I think this should be omitted.

Reply:
I did some research and we cannot remove this method. We need this method when we are trying to do the id re-assemblying (the operation-id-mapping configuration) in LRO.
The operations on the partial resource might also need this configuration to correct things, therefore we need to keep this method.
We could make this internal so that our customer do not have access to this method.